pkg/lightning/backend/external/writer.go

```diff
@@ -463,7 +463,7 @@ func (w *Writer) flushKVs(ctx context.Context, fromClose bool) (err error) {
 	var dataFile, statFile string
 	for i := 0; i < flushKVsRetryTimes; i++ {
 		dataFile, statFile, err = w.flushSortedKVs(ctx)
-		if err == nil {
+		if err == nil || ctx.Err() != nil {
```

Comment:
Normally, Can we know which errors that ctx may occur during global sort period? Dose all of them can cause tasks return?

Reply:
here I didn't want to check error. I check that user context is canceled (although the interface is `ctx.Err()`). When context is canceled, this for loop will run `flushKVsRetryTimes` times but immediately `flushSortedKVs` will return because of `ctx.Err()`, and below log line will be printed unnecessarily.